Forge.TreeWalker/src/TreeSchemaValidator.cs

```diff
@@ -0,0 +1,275 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="TreeSchemaValidator.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+// <summary>
+//     The TreeSchemaValidator class implements the ITreeSchemaValidator interface.
+// </summary>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.Serialization.Json;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Forge.DataContracts;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Schema;
+
+
+namespace Microsoft.Forge.TreeWalker
+{
+    public class TreeSchemaValidator : ITreeSchemaValidator
+    {
+        private JSchema ForgeSchemaValidationRules;
+        public TreeSchemaValidator() => ForgeSchemaValidationRules = GetRulesFromPath("contracts\\ForgeSchemaValidationRules.json");
+
+        public async Task<Tuple<bool, IList<string>>> ValidateSchema(ForgeTree schema)
```

Comment:
>ValidateSchema [](http://example.com/codeflow?start=54&length=14)

I think the methods we neeed are as follows. We need to handle ForgeTree or Dictionary<string, ForgeTree> for every form. The forms are the deserialized ForgeTree/Dictionary, the serialized string, the path, and the directory path.

Every method will take in a rules file as a string. We'll just need to figure out how to get the multi-rule-file working without having to split every method again.

ForgeTree
Dictionary<string, ForgeTree>

ForgeTree
string - serialized ForgeTree
string - path to individual ForgeSchema file
string - directory path to mulitple ForgeSchema files.

string - rules

Reply:
Perhaps to solve the rules issue, we keep what you have and accept an object. Users can pass a string rules file or a JSchema.

For the multi-rule-file case, they will create a JSchema separately and pass that in to the same method.